exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/parser/CompoundIdentifierConverter.java

```diff
@@ -115,6 +119,18 @@ public SqlNode visitChild(
           enableComplex = true;
         }
       }
+      if (expr.getKind() == SqlKind.SELECT) {
+        if (((SqlSelect) expr).getFrom() instanceof DrillCompoundIdentifier) {
+          fullSchemasSet.add((DrillCompoundIdentifier) ((SqlSelect) expr).getFrom());
+        } else if (((SqlSelect) expr).getFrom() instanceof SqlJoin) {
```

Comment:
You are right. I should add recursive checking of full-schema identifier of every nested query. 
Anyway it doesn't help now because when I ran the query with nested subqueries with different schema-qualified tables I got an error from calcite. I've already written about it in calcite dev list.